### PR TITLE
fix: fixes some bugs in artwork list creation flow

### PR DESCRIPTION
**Pull request by nickskalkin**
The type of this PR is: **Fix**

### Description

This PR fixes two issues mentioned during Artwork Lists QA

### Issue 1: Do not display second step of the model when user has no artworks

Demo:

https://user-images.githubusercontent.com/3934579/230072008-79b2b13a-e2ef-4a5a-b656-407d49af3317.mp4

### Issue 2: Clicking on save button when no artworks were selected shouldn't fail

Demo:

https://user-images.githubusercontent.com/3934579/230072020-1647fcfa-b010-423c-8452-567d68957772.mp4

### CODE REVIEW

File: src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/CreateNewListModalWizard.tsx [hunk 1]
```diff
@@ -37,3 +50,38 @@ export const CreateNewListModalWizard: FC<CreateNewListModalWizardProps> = ({
...
+          }
+        }
+      `}
+      render={({ props: relayProps, error }) => {
+        if (error) {
+          console.error(error)
+          return null
+        }
+
+        if (!relayProps?.me) {
```
Comment (dimatretyak) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/CreateNewListModalWizard.tsx:79]: In this code there is a small problem with the fact that the modal can be displayed with In this code there is a small problem with the fact that the modal can be displayed with a delay when the user has a bad Internet connection when the user has a bad Internet connection (see attached video). There are 2 ways to solve this problem:
1. Query and pass the artworks count for the `ArtworkListsHeader` in [the parent component](https://github.com/artsy/force/blob/28e7b7ae9887a4e5789c3c38e8dd88dae3356088/src/Apps/CollectorProfile/Routes/Saves2/CollectorProfileSaves2Route.tsx#L59)
2. Don't wait until a response is received from the backend and display the modal

IMHO the first way will be better 🤔


https://user-images.githubusercontent.com/3513494/230112626-0a8293cd-3379-44c4-88e5-3df26f9aa48c.mp4

File: src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/CreateNewListModalWizard.tsx [hunk 2]
```diff
@@ -1,19 +1,32 @@
...
+import { SystemQueryRenderer } from "System/Relay/SystemQueryRenderer"
+import {
+  CreateNewListModalWizardQuery,
+  CreateNewListModalWizardQuery$data,
+} from "__generated__/CreateNewListModalWizardQuery.graphql"
 
 export interface CreateNewListModalWizardProps {
   onComplete: (artworkList: ArtworkList) => void
   onClose: () => void
+  me: CreateNewListModalWizardQuery$data["me"]
```
Comment (dimatretyak) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/CreateNewListModalWizard.tsx:14]: **suggestion**: IMHO it's better to use `createFragment` here. Like this

```javascript
import { CreateNewListModalWizard_me$data } from "__generated__/CreateNewListModalWizard_me.graphql"

...

export interface CreateNewListModalWizardProps {
  ...
  me: CreateNewListModalWizard_me$data
}

...

const CreateNewListModalWizardFragmentContainer = createFragmentContainer(
  CreateNewListModalWizard,
  {
    me: graphql`
      fragment CreateNewListModalWizard_me on Me {
        collection(id: "saved-artwork") {
          artworksConnection {
            totalCount
          }
        }
      }
    `,
  }
)

...

return (
  <CreateNewListModalWizardFragmentContainer
    me={relayProps.me}
    {...props}
  />
)
```

File: src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/CreateNewListModalWizard.tsx (hunk 1 above)
Comment (nickskalkin) [reply src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/CreateNewListModalWizard.tsx:79]: thank you, good advice!

File: src/Apps/CollectorProfile/Routes/Saves2/CollectorProfileSaves2Route.tsx [hunk 3]
```diff
@@ -110,6 +114,10 @@ export const CollectorProfileSaves2RouteFragmentContainer = createFragmentContai
         allSavesArtworkList: collection(id: "saved-artwork") {
           internalID
           ...ArtworkListItem_item
+
+          savedArtworksConnection: artworksConnection {
```
Comment (dimatretyak) [inline src/Apps/CollectorProfile/Routes/Saves2/CollectorProfileSaves2Route.tsx:118]: i think you can use `artworksConnection` here

File: src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/__tests__/AddArtworksModal.jest.tsx [hunk 4]
```diff
@@ -48,26 +48,15 @@ describe("AddArtworksModal", () => {
     expect(screen.getByRole("button", { name: /Save/ })).toBeInTheDocument()
   })
 
-  it("calls the mutation when the Save button is clicked", async () => {
```
Comment (dimatretyak) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/__tests__/AddArtworksModal.jest.tsx:51]: could you please add this test case when `savedArtworksCount` is NOT 0?

**Review by dimatretyak — APPROVED** [review]
LGTM 👍

**Review by dimatretyak — APPROVED** [review]

**Pull request merged**